System.Compiler/ClrStrongName.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Runtime.InteropServices;
+
+namespace System.Compiler
+{
+    public static class ClrStrongName
+    {
+        private static IClrStrongName clrStrongName;
+        private static IClrStrongName GetClrStrongName()
+        {
+            return clrStrongName ?? (clrStrongName =
+                (IClrStrongName)RuntimeEnvironment.GetRuntimeInterfaceAsObject(
+                    new Guid("B79B0ACD-F5CD-409b-B5A5-A16244610B92"),
+                    typeof(IClrStrongName).GUID));
+        }
+
+        public static void SignatureGeneration(string filePath, string keyContainer, byte[] keyBlob)
+        {
+            GetClrStrongName().StrongNameSignatureGeneration(filePath, keyContainer, keyBlob, keyBlob.Length, IntPtr.Zero, IntPtr.Zero);
+        }
+
+        [ComImport, ComConversionLoss, InterfaceType(ComInterfaceType.InterfaceIsIUnknown), Guid("9FD93CCF-3280-4391-B3A9-96E1CDE77C8D")]
+        private interface IClrStrongName
```

Comment:
For future readers: this is a bad interface declaration because COM interface calls are made in .NET not by method name but rather method declaration order inside the interface. There are 16 missing method declarations that should have come lexically above `StrongNameSignatureGeneration`, and therefore the first slot number is used and this method actually calls `GetHashFromAssemblyFile` in spite of its name.

See the fix: https://github.com/dotnet/ILMerge/pull/49